### ISSUE TITLE
[PreferenceItem("Node Editor")] in NodeEditorReflection.cs deprecated in Unity 2019

### ISSUE THREAD

**Issue by carlotes247** [Siccity/xNode#109]
It seems that in Unity 2019 many things are different and the [PreferenceItem("Node Editor")] under NodeEditorReflection.cs throws an error because it's now deprecated. Unity apparently now uses [SettingsProvider], but the way it works is quite different. 

Currently just by changing one attribute with the other breaks the Preferences customisation menu.

More info:
https://docs.unity3d.com/ScriptReference/PreferenceItem.html
https://docs.unity3d.com/2018.3/Documentation/ScriptReference/SettingsProvider.html

**Comment by Siccity** [comment]
Which Unity version specifically? I tested with 2019.1.0b3 without issues.

**Comment by carlotes247** [comment]
I tested with 2019.1.0b2 . I will try to update to the latest beta and let you know if that solves the issue. 

I updated the Unity forum with more info about it : 	

>  Thank you for answer @Dizy we were using the master version already (latest one), but I have been trying a few things and I have new comments.
> 
>     It seems that if I create an empty Unity 2019 project and then I copy-paste the scripts from xNode, I get the same errors that I mentioned before:
> 
>     However, if I create an empty Unity 2017.3 project, copy-paste scripts from xNode, and then upgrade that project to Unity 2019 xNode seems to work. Maybe there is some sort of magic upgrade logic in the background?
> 
>     To be honest, it seems a bit weird since the same line of code that causes problems in my first approach is still there, with a green warning in Visual Studio but I can compile the project and the preferences window still appears and works.
> 
>     Any thoughts?

**Comment by carlotes247** [comment]
Updated the project that was giving the error from Unity 2019.1.0b2 to Unity 2019.1.0b3 and the error still persists and I can't compile.
![image](https://user-images.githubusercontent.com/6718809/52781550-c145d780-3044-11e9-9104-9bffdd1d5de3.png)

Still, I'm puzzled that the other xNode project I updated from Unity 2017.3 to Unity 2019.1.0b2 flags the same error as a warning and allows me to compile. 

My current hunch might be either that there is some background update logic from Unity 2017 to Unity 2019 **OR** there is an option in unity to change the way certain deprecated lines flag when compiling.
